components/istio-kyma-patch/patch.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+if [[ -z ${REQUIRED_ISTIO_VERSION} ]]; then
+    echo "Please set REQUIRED_ISTIO_VERSION variable!"
+    exit 1
+fi
+
+if [[ -z ${CONFIG_DIR} ]]; then
+    CONFIG_DIR=${DIR}
+fi
+
+function require_istio_version() {
+    local version
+    version=$(kubectl -n istio-system get deployment istio-pilot -o jsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{print $2}')
+    if [[ "$version" != ${REQUIRED_ISTIO_VERSION} ]]; then
+        echo "Istio must be in version: $REQUIRED_ISTIO_VERSION!"
+        exit 1
+    fi
+}
+
+function require_istio_system() {
+    kubectl get namespace istio-system
+}
+
+function run_all_patches() {
+  echo "--> Patch resources"
+  for f in $(find ${CONFIG_DIR} -name '*\.patch\.json' -maxdepth 1); do
+    local type=$(basename ${f} | cut -d. -f2)
+    local name=$(basename ${f} | cut -d. -f1)
+    echo "    Patch $type $name from: ${f}"
+    local patch=$(cat ${f})
+    set +e
+    local out
+    out=$(kubectl patch ${type} -n istio-system ${name} --patch "$patch" --type json)
+    local result=$?
+    set -e
+    echo "$out"
+    if [[ ${result} -ne 0 ]] && [[ ! "$out" = *"NotFound"* ]]; then
+        exit ${result}
+    fi
+  done
+}
+
+function remove_not_used() {
+  echo "--> Delete resources"
+  while read line; do
+    echo "    Delete $line"
+    local type=$(cut -d' ' -f1 <<< ${line})
+    local name=$(cut -d' ' -f2 <<< ${line})
+    set +e
+    kubectl delete ${type} ${name} -n istio-system
+    set -e
+  done <${CONFIG_DIR}/delete
+}
+
+function configure_sidecar_injector() {
+  echo "--> Configure sidecar injector"
+  local configmap=$(kubectl -n istio-system get configmap istio-sidecar-injector -o jsonpath='{.data.config}')
+  local alreadyEnabled=$(grep "policy: enabled" <<< "$configmap")
+  if [[ -n ${alreadyEnabled} ]]; then
+    # Disable automatic injecting
+    configmap=$(sed 's/policy: enabled/policy: disabled/' <<< "$configmap")
+    configmap=$(sed 's/\[\[ \.ProxyConfig\.ZipkinAddress \]\]/zipkin.kyma-system:9411/g' <<< "$configmap")
+
+    # Set limits for sidecar. Our namespaces have resource quota set thus every container needs to have limits defined.
+    # Add limits to already existing resources sections
+    configmap=$(sed 's|    resources:|    resources:\'$'\n      limits: { memory: 128Mi, cpu: 100m }|' <<< "$configmap")
```

Comment:
why u are not specifying `requests` property?

Reply:
We are setting limits here, because our namespaces have resource-quota. In such case every container needs to have limits set. Usually it is done by limit-range, but it does not apply to injected containers. Setting requests is not necessary so we skipped that part and used what they provided.

Reply:
"Setting requests is not necessary so we skipped (..)" but why it's not necessary. IMO you should because it will not be verbose enough and you rely on the internal implementation of k8s. Please set it as it was before.

Reply:
Previously `requests.memory` for istio-proxy was set to same value as `limits.memory` in [istio/values.yaml](https://github.com/kyma-project/kyma/blob/master/resources/istio/values.yaml#L36). According to [k8s docs](https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-default-namespace/#what-if-you-specify-a-container-s-limit-but-not-its-request) it is the same as not specifying requests at all. So, accidentally we have the same behaviour as previously :).

Reply:
so, as I said: "it will not be verbose enough"  and you can see my first comment, "By default, the request will be the same as the limit." but this requires additional knowledge and can be changed in future so you also rely on the k8s implementation.

what's more the request for [CPU](https://github.com/kyma-project/kyma/blob/master/resources/istio/values.yaml#L35) was different, so it's not the same as previously

Reply:
fixed